scripts/apps/monitoring/controllers/AggregateCtrl.ts

```diff
@@ -338,55 +344,81 @@ export function AggregateCtrl($scope, desks, workspaces, preferencesService, sto
      */
     this.hasFileType = function(fileType) {
         if (fileType === 'all') {
-            return this.selectedFileType.length === 0;
+            return this.activeFilters.fileType.length === 0;
         }
-        return this.selectedFileType.indexOf(fileType) > -1;
+        return this.activeFilters.fileType.includes(fileType);
     };
 
     /**
      * Return selected file types if the 'fileType' filter(s) is selected
      * @return [{string}] fileType
      */
     this.getSelectedFileTypes = function() {
-        return this.selectedFileType.length === 0 ? null : JSON.stringify(this.selectedFileType);
+        return this.activeFilters.fileType.length === 0 ? null : JSON.stringify(this.this.activeFilters.fileType);
     };
 
-    /**
-     * Update the type filter criteria
-     */
-    function updateFileTypeCriteria() {
-        var value = self.selectedFileType.length === 0 ? null : JSON.stringify(self.selectedFileType);
+    function updateFilteringCriteria() {
+        _.forEach(self.activeFilters, (filterValue, filterType) => {
+            var value = filterValue.length === 0 ? null : JSON.stringify(filterValue);
 
-        _.each(self.groups, (item) => {
-            item.fileType = value;
-        });
-        _.each(self.spikeGroups, (item) => {
-            item.fileType = value;
+            _.each(self.groups, (item) => {
+                item[filterType] = value;
+            });
+            _.each(self.spikeGroups, (item) => {
+                item[filterType] = value;
+            });
         });
     }
 
-    /**
-     * Set the current 'fileType' filter
-     * param {string} fileType
-     */
-    this.setFileType = function(fileType) {
-        if (fileType === 'all') {
-            this.selectedFileType = [];
-        } else {
-            var index = this.selectedFileType.indexOf(fileType);
-
-            if (index > -1) {
-                this.selectedFileType.splice(index, 1);
+    this.setFilterType = function(filterType, filterValue, $event?) {
+        if (filterType === 'profile') {
+            if (!this.activeFilters.profile.includes(filterValue._id)) {
```

Comment:
It would be clearer if it was named `activeFilters.contentProfile` instead of just `profile`, because I was thinking about user profiles.